README.md

```diff
@@ -139,6 +139,39 @@ only really makes sense if you're calling Grover directly (and not via middlewar
 Grover.new('<some URI with basic authentication', username: 'the username', password: 'super secret').to_pdf
 ```
 
+#### Adding cookies
+If you need to set cookies to your new page instance, just pass them along like this:
+
+```ruby
+myCookies = [{:name=>"sign_username", :value=>"any@any.com", :domain=>"mydomain"}, {:name=>"_session_id", :value=>"9c014df0b699d8dc08d1c472f8cc594c", :domain=>"mydomain"}]
+Grover.new('<some URI with cookies', cookies: myCookies ).to_pdf
+```
+
+If, let's say, you wanted to just forward the cookies that you got on your original request, you can do something like:
+
+```ruby
+def header_cookies(req=nil)
+    req ||= request
+    cookie_dough = request.headers['Cookie'].split '; '
+    resultHash = Hash[cookie_dough.map do |c| 
+      pieces = c.split '='
+      [pieces[0].to_sym, pieces[1]]
+    end]
+
+    return resultHash.each_with_object([]) do |(key, value), array|
+      array << { name: key, value: value, domain: request.headers["Host"] }
+    end
+  end
```

Comment:
Doesn't look like you were using `req` and iterating over the cookies twice didn't seem necessary. Something like this would likely do fine..

```suggestion
def header_cookies
  request.headers['Cookie'].split('; ').map do |cookie|
    key, value = cookie.split '='
    { name: key, value: value, domain: request.headers['Host'] }
  end
end
```

Reply:
I tested it and it is working so I'll commit your suggestion. I'm not a Rails developer and this is my first pull request ever so you can guess :smile: